### ISSUE TITLE
Bug with external HDF5 links

### ISSUE THREAD

**Issue by woutdenolf** [silx-kit/jupyterlab-h5web#39]
When the top-level HDF5 groups are external links (BLISS proposal and sample collection files), the entire file doesn't show up and this traceback is printed:

```python
[E 2021-06-16 23:54:31.907 ServerApp] Found and opened file, error getting contents from object specified by the uri.
    Error: Traceback (most recent call last):
      File "/users/denolf/virtualenvs/jupyter/ubuntu_20_04/lib/python3.9/site-packages/jupyterlab_hdf/baseHandler.py", line 72, in get
        out = self._get(fpath, uri, **kwargs)
      File "/users/denolf/virtualenvs/jupyter/ubuntu_20_04/lib/python3.9/site-packages/jupyterlab_hdf/baseHandler.py", line 90, in _get
        return self._getFromFile(f, uri, **kwargs)
      File "/users/denolf/virtualenvs/jupyter/ubuntu_20_04/lib/python3.9/site-packages/jupyterlab_hdf/meta.py", line 16, in _getFromFile
        return jsonize(hobjMetaDict(f[uri], ixstr=ixstr, min_ndim=min_ndim))
      File "h5py/_objects.pyx", line 54, in h5py._objects.with_phil.wrapper
      File "h5py/_objects.pyx", line 55, in h5py._objects.with_phil.wrapper
      File "/users/denolf/virtualenvs/jupyter/ubuntu_20_04/lib/python3.9/site-packages/h5py/_hl/group.py", line 288, in __getitem__
        oid = h5o.open(self.id, self._e(name), lapl=self._lapl)
      File "h5py/_objects.pyx", line 54, in h5py._objects.with_phil.wrapper
      File "h5py/_objects.pyx", line 55, in h5py._objects.with_phil.wrapper
      File "h5py/h5o.pyx", line 190, in h5py.h5o.open
    KeyError: "Unable to open object (object '8.1' doesn't exist)"
    
    relfpath: data/inhouse/id002106/id00/id002106_id00.h5, uri: /8.1, min_ndim: None, ixstr: None, subixstr: None
```

The HDF5 group of the link destination is indeed `/8.1` but the `relfpath` is the path of the file that contains the link, not the path of the file of the link destination.

```python
>>> import os, h5py
>>> os.getcwd()
'/tmp/data/inhouse/id002106/id00'

>>> f = h5py.File("id002106_id00.h5")
>>> f.filename
'id002106_id00.h5'

>>> f["sample_0005_8.1"].file.filename
'/tmp/data/inhouse/id002106/id00/sample/sample_0005/sample_0005.h5'
>>> f["sample_0005_8.1"].name
'/8.1'

>>> link = f.get("sample_0005_8.1", getlink=True)
>>> link.filename
'sample/sample_0005/sample_0005.h5'
>>> link.path
'/8.1'
```

The jupyterlab working directory was `/tmp`

**Comment by woutdenolf** [comment]
Minimal files to illustrate the issue:

```python
import h5py

with h5py.File("data/destfile.h5", "w") as f:
    g = f.create_group("destgroup")
    g["data"] = [1, 2, 3]

with h5py.File("data/sourcefile.h5", "w") as f:
    f["sourcegroup"] = h5py.ExternalLink("destfile.h5", "destgroup")
```

Opening the "data/sourcefile.h5" will show the issue.

**Comment by loichuder** [comment]
Fixed by the new h5grove-based backend #40
